packages/loader/container-loader/src/loader.ts

```diff
@@ -276,10 +276,22 @@ export class Loader extends EventEmitter implements ILoader {
     public async createDetachedContainer(codeDetails: IFluidCodeDetails): Promise<Container> {
         debug(`Container creating in detached state: ${performance.now()} `);
 
-        return Container.createDetached(
+        const container = await Container.createDetached(
             this,
             codeDetails,
-            );
+        );
+
+        if (this.cachingEnabled) {
+            container.once("attached", () => {
+                ensureFluidResolvedUrl(container.resolvedUrl);
```

Comment:
probably want a closed check before we cache. also some tests would be good.

Reply:
Added closed check (and moved up the caching before the await) in latest.  I think this test covers the scenario already: https://github.com/microsoft/FluidFramework/blob/main/packages/test/end-to-end-tests/src/test/detachedContainerTests.spec.ts#L628